test/libsolidity/semanticTests/freeFunctions/free_namesake_contract_function.sol

```diff
@@ -5,7 +5,11 @@ contract C {
   }
 }
 // ====
-// compileViaYul: also
 // compileToEwasm: also
+// compileViaYul: also
 // ----
 // f() -> FAILURE
+// gas ir: 100000000
+// gas legacy: 100000000
+// gas legacyOptimized: 100000000
+// gas irOptimized: 100000000
```

Comment:
Is this one so expensive because the failure eats all remaining gas?

Reply:
Looks like that